state-chain/primitives/src/chains/assets.rs

```diff
@@ -531,6 +533,17 @@ macro_rules! assets {
 	}
 }
 
+/// Returns an amount for an none-gas asset we use for transaction fee estimation.
+pub fn get_simulation_amount(asset: Asset) -> Option<u128> {
```

Comment:
I don't think this needs to be in the primitives crate. It's not a primitive and it's not used anywhere except the ingress/egress pallet. (in fact not sure it even needs to be a function).